TruffleString-Experiment.md

```diff
@@ -201,21 +198,21 @@ The table indicates that the second experiment still shows a performance decreas
 
-| Metric           | Tier | Baseline      | Experiment 1  | Experiment 2  | Change (%) |
-| ---------------- | ---- | ------------- | ------------- | ------------- | ---------- |
-| Compilation Time | 1    | 6 948 955 ms  | 8 333 508 ms  | 8 162 938 ms  | -2.1% ⬇️    |
-| Compilation Time | 2    | 10 044 058 ms | 21 799 170 ms | 23 663 382 ms | +8.5% ⬆️    |
-| Code Size        | 1    | 304 688 bytes | 325 563 bytes | 325014 bytes  | -0.2% ⬇️    |
-| Code Size        | 2    | 742 502 bytes | 946 422 bytes | 1000615 bytes | +5.3% ⬆️    |
+| Metric           | Tier | Baseline      | Experiment 1  | Experiment 2    | Change to Baseline (%) | Change to Experiment 1 (%) |
+| ---------------- | ---- | ------------- | ------------- | --------------- | ---------------------- | -------------------------- |
+| Compilation Time | 1    | 6 948 955 ms  | 8 333 508 ms  | 8 162 938 ms    | +17.5% ⬆️               | -2.1% ⬇️                    |
+| Compilation Time | 2    | 10 044 058 ms | 21 799 170 ms | 23 663 382 ms   | +136.5% ⬆️              | +8.5% ⬆️                    |
+| Code Size        | 1    | 304 688 bytes | 325 563 bytes | 325 014 bytes   | +6.7% ⬆️                | -0.2% ⬇️                    |
+| Code Size        | 2    | 742 502 bytes | 946 422 bytes | 1 000 615 bytes | +34.5% ⬆️               | +5.3% ⬆️                    |
 
 The compilation statistics show an improvement of code size and compilation time in Tier 1, but an increase in Tier 2. It is still larger in comparison to the baseline performance, which indicates that the compiler had to do more work to optimize the code. This may be, because we replace byte operations with TruffleString operations. These are represented as nodes, which are more complex and require more compilation time.
 
 The goal of the second experiment was to reduce the code size of the `String>>#compareWith:` method. The code size of the `String>>#compareWith:` method for the second experiment is as follows:
 
-| Metric    | Tier | Baseline    | Experiment 1 | Experiment 2  | Change (%) |
-| --------- | ---- | ----------- | ------------ | ------------- | ---------- |
-| Code Size | 1    | 2 040 bytes | 10 840 bytes | 8 276 bytes ⬆️ | -23.6% ⬇️   |
-| Code Size | 2    | 897 bytes   | 6 839 bytes  | 4 403 bytes   | -36.0% ⬇️   |
+| Metric    | Tier | Baseline    | Experiment 1 | Experiment 2 | Change to Baseline (%) | Change to Experiment 1 (%) |
+| --------- | ---- | ----------- | ------------ | ------------ | ---------------------- | -------------------------- |
+| Code Size | 1    | 2 040 bytes | 10 840 bytes | 8 276 bytes  | +405.7% ⬆️              | -23.6% ⬇️                   |
+| Code Size | 2    | 897 bytes   | 6 839 bytes  | 4 403 bytes  | +491.4% ⬆️              | -36.0% ⬇️                   |
```

Comment:
Those numbers compared to the baseline are pretty wild XD

Reply:
But real 😢